test/b-splines/cubic.jl

```diff
@@ -0,0 +1,48 @@
+# module CubicTests
+
+using Base.Test
+using Interpolations
+
+for (constructor, copier) in ((interpolate, identity), (interpolate!, copy))
+    f(x) = sin((x-3)*2pi/9 - 1)
```

Comment:
IMO it would be great to use a cubic polynomial for testing, too, so you can be sure your values are exact. In the LaTeX docs, see the section about `InPlaceQ` to be sure the boundary conditions aren't destroying the exact equality.